### PR TITLE
[content-service] Implement UsageReportService.DownloadURL

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Implements DownloadURL rpc. This is needed to be able to download a usage report form the usage component.

## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/12300

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: components/content-service/pkg/storage/storage.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ const (
 
 var (
 	// ErrNotFound is returned when an object is not found
-	ErrNotFound = xerrors.Errorf("not found")
+	ErrNotFound = fmt.Errorf("not found")
```
Comment (easyCZ) [inline components/content-service/pkg/storage/storage.go:35]: Drive-by: `xerrors.Errorf` is deperacted

File: components/content-service/pkg/service/usage-report-service.go [hunk 2]
```diff
@@ -42,20 +42,64 @@ func (us *UsageReportService) UploadURL(ctx context.Context, req *api.UsageRepor
 	span.SetTag("name", req.Name)
 	defer tracing.FinishSpan(span, &err)
 
-	err = us.s.EnsureExists(ctx, us.bucketName)
+	if req.GetName() == "" {
+		return nil, status.Error(codes.InvalidArgument, "Name is required but got empty.")
+	}
+
+	logger := log.WithField("name", req.Name).
```
Comment (sagor999) [inline components/content-service/pkg/service/usage-report-service.go:49]: usually we just name the variable `log` in cases like this. 
```suggestion
	log := log.WithField("name", req.Name).
```

**Review by sagor999 — APPROVED** [review]

File: components/content-service/pkg/service/usage-report-service.go (hunk 2 above)
Comment (easyCZ) [reply components/content-service/pkg/service/usage-report-service.go:49]: `log` is the package name where the logging is implemented. It's a name clash that we shouldn't allow.

**Comment by easyCZ** [comment]
dependency landed
/unhold
